emukit/quadrature/methods/bounded_bq_model.py

```diff
@@ -0,0 +1,136 @@
+            raise ValueError(f"{self.__class__.__name__} can only be used with QuadratureRBFIsoGaussMeasure kernel. "
+                             f"Instead {type(base_gp.kern)} is given.")
+
+        super(BoundedBayesianQuadrature, self).__init__(base_gp=base_gp,
+                                                        warping=SquareRootWarping(
+                                                                 offset=bound, is_inverted=not is_lower_bounded
+                                                             ),
+                                                        X=X,
+                                                        Y=Y)
+
+    @property
+    def bound(self):
+        """The bound :math:`\alpha` as defined in the model. The true bound of the integrand might be different."""
+        return self._warping.offset
+
+    @property
+    def is_lower_bounded(self):
+        """``True`` if the model is lower bounded, ``False`` if it is upper bounded."""
+        return not self._warping.is_inverted
+
+    def predict_base(self, X_pred: np.ndarray) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+        """Compute the predictive mean and variance of the warped GP as well as the base GP.
+
+        :param X_pred: Locations at which to predict, shape (num_points, input_dim).
+        :returns: Predictive mean and variances of warped GP :math:`\\hat{f}`, and predictive mean and variances
+                  of base-GP :math:`g` in that order, all shapes (num_points, 1).
+        """
+        mean_base, var_base = self.base_gp.predict(X_pred)
+
+        mean_approx = self.transform(mean_base)
+        var_approx = var_base * (mean_base ** 2)
+        return mean_approx, var_approx, mean_base, var_base
+
+    def predict_base_with_full_covariance(self, X_pred: np.ndarray) -> Tuple[np.ndarray, np.ndarray, np.ndarray,
+                                                                             np.ndarray]:
+        """Compute predictive mean and covariance of the warped GP as well as the base GP.
+
+        :param X_pred: Locations at which to predict, shape (num_points, input_dim)
+        :returns: Predictive mean and covariance of warped GP :math:`\\hat{f}`, predictive mean and covariance of
+                  base-GP :math:`g` in that order. Mean shapes both (num_points, 1) and covariance shapes both
+                  (num_points, num_points)
+        """
+        mean_base, cov_base = self.base_gp.predict_with_full_covariance(X_pred)
+
+        mean_approx = self.transform(mean_base)
+        cov_approx = np.outer(mean_base, mean_base) * cov_base
+        cov_approx = self.symmetrize_matrix(cov_approx)  # for numerical stability
+        return mean_approx, cov_approx, mean_base, cov_base
+
+    def integrate(self) -> Tuple[float, float]:
+        """Compute the normal distribution of the integral value, in particular its mean estimator and variance.
```

Comment:
I canged `N` to `n_poins` and `D` to `n_input_dim` to mimic the docs and `W` to `Weights_outer` since it is the outer product matrix of the weights. `K` should be fine as it is the kernel matrix and we use that naming elsewhere.